app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,17 @@
+class LikesController < ApplicationController
+  def new
+    @user = User.find(params[:user_id])
+  end
+
+  def create
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @like = @post.likes.new
+    @like.author = @user
+    if @like.save
+      redirect_to user_post_path(@user, @post)
```

Comment:
- [ ] Please make sure to use the created `current_user` method in the application_controller file as per the project requirements. This will ensure that the user is accessible easily by the controllers.
`@like = @current_user.likes.new(post_id: @post.id)`